cache.py

```diff
@@ -0,0 +1,66 @@
+from collections import OrderedDict
+from collections import deque
+import sys
+import os
+#from cachetools import *
+from cachemonCache import S3FIFO
+
+total = 0
+class _LRUCache:
+    def __init__(self, capacity: int):
+        self.hit = 0
+        self.capacity = capacity
+        self.cache = OrderedDict()
+
+    def get(self, key: int) -> int:
+        if key in self.cache:
+            self.cache.move_to_end(key)
+            return self.cache[key]
+        return -1
+
+    def put(self, key: int, value: int) -> None:
+        if key in self.cache:
+            self.hit += 1
+            self.cache[key] = value
+            self.cache.move_to_end(key)
+        else:
+            if len(self.cache) >= self.capacity:
+                self.cache.popitem(last=False)
+            self.cache[key] = value
+
+
```

Comment:
```suggestion
class GraphBoltS3FIFO:
    def __init__(self, number_of_elements: int, num_parts: int):
        self.cache = gb.impl.FeatureCache([number_of_elements, 1], torch.int64, num_parts)

    def get(self, key: int) -> int:
        values, missing_index, missing_keys = self.cache.query(torch.tensor([key]))
        if missing_keys.numel() == 0:
            return values[0].item()
        return -1

    def put(self, key: int, value: int) -> None:
        self.cache.replace(torch.tensor([key]), torch.tensor([value]))

    @property
    def hit_rate(self) -> float:
        return 1 - self.cache.miss_rate
```

Reply:
num_parts is there to add parallelism. It would be nice if we could measure num_parts == 1 and num_parts == 16.

Reply:
i see. thank you for your advice!